src/components/send/form/form.js

```diff
@@ -1,4 +1,10 @@
 // eslint-disable-line max-lines
+// TODO Split this Send form component into following 4:
+// - FormWrapper - decides to render FormLSK or FormBTC based on activeToken
+// - FormLSK - Contains all LSK-specific functionality (message, static fee) and renders FormBase
+// - FormBTC - Contains all BTC-specific functionality (dynamic fee selection) and renders FormBase
+// - FormBase - Contains all functionality common to LSK and BTC (Box, address input,
+//     amount input, confirm button)
```

Comment:
@slaweet I understand the component for FormLSK and FormBTC but the FormBase looks like the the parent component and then I don't understand the FormWrapper.

If I understand right will be something like.

- FormBase
   - FormWrapper
      - FormLSK/FormBTC

In any case I think that most important is the FormLSK and FormBTC what will be very good for the send.

Reply:
What I meant is something like this (very simplified):

```
const FormWrapper = ({ activeToken, ...props }) => (
  activeToken === 'BTC' ? <FormBTC {...props} /> : <FormLSK {...props} />
);
```
```
const FormLSK = (props) => (
  <FormBase {...props} >
    <StaticFee />
    <MessageInput />
  </FormBase>
);
```
```
const FormBTC = (props) => (
  <FormBase {...props} >
    <DynamicFee />
  </FormBase>
);
```
```
const FormBase = ({ children }) => (
  <Box>
    <AddressInput />
    <AmountInput />
     {children}
   <ActionButtons />
  </Box>
);
```

I used a similar pattern in Request https://github.com/LiskHQ/lisk-hub/blob/development/src/components/request but there, `RequestWrapper` corresponds to `FormBase` in this proposal and `index.js`(`Request`) is proposed `FormWrapper`. So I would change the naming here to be consistent with the request.

Reply:
I'd go with dividing the components based on their purpose rather than the values they use. An amount input, is gonna validate the amount and show the fiat equivalent not matter the token. the difference will only be the pure validation function.
Same goes for recipient input. Address validation and fetching the bookmarks is exactly the same. I'd suggest not duplicate this functionality in two components (FormBTC and FormLSK). Instead, better have it central with a pure validation function dedicated for each. I assume with proper architecture, fetching bookmarks must be no different other than passing different parameters to the same function.
On thing is to decide to show or hide the Message, which is as simple as it can get.

Reply:
@reyraa I think you miss-read what I proposed above. 

AddressInput and AmountInput is in the FormBase component, so it's common for BTC and LSK. 

What is not described above is how they get the token-specific validation functions. Token-specific component can easily pass a token-specific function for that into the FormBase, like `validateAddress` below:

```
const FormBase = ({ children, validateAddress }) => (
  <Box>
    <AddressInput validateAddress={validateAddress} />
    <AmountInput />
     {children}
   <ActionButtons />
  </Box>
);
```

Reply:
And also I didn't mean that `AddressInput` and `AmountInput` necessarily need to be separate components (depends on their size). I just put them to illustrate what functionality will be in which component.

Reply:
Ok, I got you now. I agree with what proposed then.

Reply:
I think it is better to have on parent form and define the differences inside the form component like `AddressInput` and `AmountInput`. Then you can move the validation methods to a utility.